git_machete/cli.py

```diff
@@ -75,274 +77,423 @@ def usage(command: str = None) -> None:
         """[1:])))
 
 
-def short_usage() -> None:
-    print(
-        fmt("<b>Usage: git machete [--debug] [-h] [-v|--verbose] [--version] "
-            "<command> [command-specific options] [command-specific argument]</b>"))
+def short_usage() -> str:
+    return (fmt("<b>Usage: git machete [--debug] [-h] [-v|--verbose] [--version] "
+                "<command> [command-specific options] [command-specific argument]</b>"))
 
 
 def version() -> None:
     print(f"git-machete version {__version__}")
 
 
-def main() -> None:
-    launch(sys.argv[1:])
-
-
-def launch(orig_args: List[str]) -> None:
-    cli_opts = git_machete.options.CommandLineOptions()
-    git = GitContext(cli_opts)
-
-    if sys.version_info.major == 2 or (sys.version_info.major == 3 and sys.version_info.minor < 6):
+class MacheteHelpAction(argparse.Action):
```

Comment:
AFAICS `-h` still awkwardly shows up in error messages

```
$ git machete  show foo
usage: git-machete show [-h]
                        {c,current,d,down,f,first,l,last,n,next,p,prev,r,root,u,up}
                        [branch]
git-machete show: error: argument direction: invalid choice: 'foo' (choose from 'c', 'current', 'd', 'down', 'f', 'first', 'l', 'last', 'n', 'next', 'p', 'prev', 'r', 'root', 'u', 'up')
```

Any chance to get rid of that?.... can be a separate issue ofc as I assume it either requires low-level hacking or is non-trivial to do properly

Reply:
There few things we can do and I will try to present their consequences in the following format show_parser definition -> result when you call `git machete show foo` .

Option A (leave as is):
`show_parser = subparsers.add_parser('show', add_help=False)`
result:
```
➜  git-machete git:(improve/parser) ✗ git machete show foo   
usage: git-machete show [-h] {c,current,d,down,f,first,l,last,n,next,p,prev,r,root,u,up} [branch]
git-machete show: error: argument direction: invalid choice: 'foo' (choose from 'c', 'current', 'd', 'down', 'f', 'first', 'l', 'last', 'n', 'next', 'p', 'prev', 'r', 'root', 'u', 'up')
```

Option B (we remove usage from displaying):
`show_parser = subparsers.add_parser('show', usage=argparse.SUPPRESS, add_help=False)`
result:
```
➜  git-machete git:(improve/parser) ✗ git machete show foo
git-machete show: error: argument direction: invalid choice: 'foo' (choose from 'c', 'current', 'd', 'down', 'f', 'first', 'l', 'last', 'n', 'next', 'p', 'prev', 'r', 'root', 'u', 'up')
```

Option C (we change usage into long_docs(command), by calling get_help_description/usage(command))
Please notice ugle double usage as the begging of the message. This can be hidden in some cases, but for stuff liek `format` it won't work:
`show_parser = subparsers.add_parser('show', usage=get_help_description('show'), add_help=False)`
result:
```
➜  git-machete git:(improve/parser) ✗ git machete show foo
usage: 
Usage: git machete show <direction> [<branch>]
where <direction> is one of: c[urrent], d[own], f[irst], l[ast], n[ext], p[rev], r[oot], u[p]
displayed relative to given <branch>, or the current checked out branch if <branch> is unspecified.

Outputs name of the branch (or possibly multiple branches, in case of down) that is:

* current: the current branch; exits with a non-zero status if none (detached HEAD)
* down:    the direct children/downstream branch of the given branch.
* first:   the first downstream of the root branch of the given branch (like root followed by next), or the root branch itself if the root has no downstream branches.
* last:    the last branch in the definition file that has the same root as the given branch; can be the root branch itself if the root has no downstream branches.
* next:    the direct successor of the given branch in the definition file.
* prev:    the direct predecessor of the given branch in the definition file.
* root:    the root of the tree where the given branch is located. Note: this will typically be something like develop or master, since all branches are usually meant to be ultimately merged to one of those.
* up:      the direct parent/upstream branch of the given branch.
git-machete show: error: argument direction: invalid choice: 'foo' (choose from 'c', 'current', 'd', 'down', 'f', 'first', 'l', 'last', 'n', 'next', 'p', 'prev', 'r', 'root', 'u', 'up')
```

If any of these options is satisfying for you, please let me know. I recommend option B, so silence it and if someone wants a help for it they should execute `git machete <command> --help`.

Reply:
Ok! let's go for B, it'd be probably even more consistent since currently:
* for `github`, `go`, `list`, `show` &mdash; if no param is provided, help is displayed
* for all other commands &mdash; invocation with no params is a valid one